src/Connection.php

```diff
@@ -166,6 +167,12 @@ class Connection
 
     private SchemaManagerFactory $schemaManagerFactory;
 
+    private static bool $isChecking = false;
+
+    private static ?int $lastCheckedAt = null;
+
+    private ?int $checkFrequency;
```

Comment:
frequency is measured in hertz (1/seconds), but you need seconds here :)
what about just `heartbeat`?